examples/05_stable_diffusion/src/compile_lib/compile_vae.py

```diff
@@ -22,62 +22,103 @@
 from .util import mark_output
 
 
-def map_vae_params(ait_module, pt_module):
-    pt_params = dict(pt_module.named_parameters())
-    mapped_pt_params = {}
-    for name, _ in ait_module.named_parameters():
-        ait_name = name.replace(".", "_")
-        if name in pt_params:
-            if (
-                "conv" in name
-                and "norm" not in name
-                and name.endswith(".weight")
-                and len(pt_params[name].shape) == 4
-            ):
-                mapped_pt_params[ait_name] = torch.permute(
-                    pt_params[name], [0, 2, 3, 1]
-                ).contiguous()
-            else:
-                mapped_pt_params[ait_name] = pt_params[name]
-        elif name.endswith("attention.proj.weight"):
-            prefix = name[: -len("attention.proj.weight")]
-            pt_name = prefix + "proj_attn.weight"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.proj.bias"):
-            prefix = name[: -len("attention.proj.bias")]
-            pt_name = prefix + "proj_attn.bias"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.cu_length"):
-            ...
-        elif name.endswith("attention.proj_q.weight"):
-            prefix = name[: -len("attention.proj_q.weight")]
-            pt_name = prefix + "query.weight"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.proj_q.bias"):
-            prefix = name[: -len("attention.proj_q.bias")]
-            pt_name = prefix + "query.bias"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.proj_k.weight"):
-            prefix = name[: -len("attention.proj_k.weight")]
-            pt_name = prefix + "key.weight"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.proj_k.bias"):
-            prefix = name[: -len("attention.proj_k.bias")]
-            pt_name = prefix + "key.bias"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.proj_v.weight"):
-            prefix = name[: -len("attention.proj_v.weight")]
-            pt_name = prefix + "value.weight"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
-        elif name.endswith("attention.proj_v.bias"):
-            prefix = name[: -len("attention.proj_v.bias")]
-            pt_name = prefix + "value.bias"
-            mapped_pt_params[ait_name] = pt_params[pt_name]
+def torch_dtype_from_str(dtype: str):
+    if dtype == "float16":
+        torch_dtype = torch.float16
+    elif dtype == "float32":
+        torch_dtype = torch.float32
+    else:
+        raise ValueError("dtype not supported yet!")
+    return torch_dtype
+
+def map_vae(pt_module, device="cuda", dtype="float16"):
```

Comment:
Is it possible that we could simplify this method a bit? (sorry I know the complex logic is probably existing before this PR, but highly appreciate if you can help simplify as you doing the change in this PR)